packages/sanity/src/core/form/studio/assetSource/DefaultSource.tsx

```diff
@@ -14,8 +14,51 @@ const PER_PAGE = 200
 const ASSET_TYPE_IMAGE = 'sanity.imageAsset'
 const ASSET_TYPE_FILE = 'sanity.fileAsset'
 
-const buildQuery = (start = 0, end = PER_PAGE, assetType = ASSET_TYPE_IMAGE) => `
-  *[_type == "${assetType}"] | order(_updatedAt desc) [${start}...${end}] {
+const buildFilterQuery = (acceptParam: string) => {
+  const WILDCARD_ACCEPT = ['image/*', 'audio/*', 'video/*']
+  const acceptItems = acceptParam.split(',').map((accept) => accept.trim())
+
+  const typesForFilter: {[key: string]: string} = acceptItems.reduce(
+    (acceptTypes: {[key: string]: string}, acceptR: string) => {
+      // builds the wildcard part of the groq query
+      if (WILDCARD_ACCEPT.includes(acceptR)) {
+        const wild = `mimeType match '${acceptR}' || ${acceptTypes.wildcards}`
+        return {...acceptTypes, ...{wildcards: wild}}
```

Comment:
The last spread here isn't necessary, and I think I would prefer the variable to be inlined for better readability:

```suggestion
{
  ...acceptTypes,
  wildcards: `mimeType match '${acceptR}' || ${acceptTypes.wildcards}`
}
```

Reply:
I think it still needs the spreader because it needs to merge "two objects" together. without the spreader the editor is telling me that "property assignement is missing". Unless I missed something? 🤔